source/how-tos/support_ticket.rst

```diff
@@ -0,0 +1,132 @@
+      email:
+        # Optional.
+        # Text/HTML added to the page under the support ticket header
+        description: |
+          <strong>My Description Text</strong><br />
+          This is a description text for the frontend with HTML support
+        # Optional.
+        # To set the email from field. Defaults to the email set in the form.
+        from: "config@example.com" 
+        # Required.
+        # Your support system email address. The email will be sent to this address
+        to: "support@support.system.com"
+        # delivery_method and delivery_settings are Rails configuration items
+        # that can be set in this file to simplify configuration.
+        # If you prefer, you can configure Rails in the usual way.
+        delivery_method: "smtp"
+        delivery_settings:
+          address: 'smtp.gmail.com'
+          port: 587
+          domain: 'example.com'
+          user_name: '<username>'
+          password: '<password>'
+          authentication: 'plain'
+          enable_starttls_auto: true
+          open_timeout: 5
+          read_timeout: 5
+
+Customizations
+..............
+
+The support ticket form can be customized using the ``form`` and ``attributes`` configuration properties.
+In the same way that you can configure the :ref:`form for interactive applications <enable-desktops-modify-form-attributes>`,
+the support ticket form fields can be arranged or changed as required.
+
+.. note::
+  
+  Please note that an ``email`` field must be present in the form for the support ticket feature to be funtional.
+
+Custom fields can be added, but they will require a custom email template to make use of the field values to generate the email body.
+Drop the template file into ``/etc/ood/config/apps/dashboard/views/support_ticket/email`` and reference it
+in the configuration. This template file can be any format Rails recognizes for emails, usually the ``.txt.erb`` extension.
+
+The example below shows a custom template configuration with 3 fields and how to use them in the template to generate the email body.
+
+  .. code-block:: yaml
+
+    support_ticket:
+      email_template: "custom_template"
+      attributes:
+        email:
+          required: true
+        subject:
+          value: "OnDemand Problem: "
+      form:
+        - email
+        - subject
+        - department
+
+  .. code-block::
+
+    # /etc/ood/config/apps/dashboard/views/support_ticket/email/custom_template.txt.erb
+    Email Intro
+    Department: <%= @context.support_ticket.department %>
+    Subject Copy: <%= @context.support_ticket.subject %>
+
+    Other static data.
+
+.. figure:: /images/support_ticket_custom_form.png
+   :align: center
+
+Troubleshooting
+...............
+
+.. tip::
+
+  Ensure that your Dashboard Rails environment is configured properly to send emails and to raise errors.
+  Update your environment settings accordingly ``dashboard/config/environments/(development.rb|test.rb|production.rb)``
```

Comment:
What is this? Real sites should not be modifying these files.

Reply:
This is only needed if there are problems with the integration and requires debugging in a local environment.
In the `development` environment, Rails `mailer` will not raise any errors, so if admins want to debug issues locally, they need to set this settings.

I added this section as it helped me debug issues while developing the feature.
Shall I remove it?

Reply:
I see. I don't know if any folks use the development environment. Usually they just deploy on a test system and interact with the production system.

> In the development environment, Rails mailer will not raise any errors, 

We should just reconfigure `development.rb` upstream so that it does.

> Shall I remove it?

Yes please.

Reply:
OK. Will remove this section and add the changes to the `development.rb` to raise errors for emails.

Reply:
removed